pulsar-client/src/main/java/org/apache/pulsar/client/impl/ConsumerBase.java

```diff
@@ -490,6 +491,15 @@ public void reconsumeLaterCumulative(Message<?> message, long delayTime, TimeUni
         if (null != txn) {
             checkArgument(txn instanceof TransactionImpl);
             txnImpl = (TransactionImpl) txn;
+           if (txnImpl.getState() != TransactionImpl.State.OPEN) {
+               CompletableFuture<Void> completableFuture = new CompletableFuture<>();
+               completableFuture
+                       .completeExceptionally(new TransactionCoordinatorClientException
+                               .InvalidTxnStatusException("["+ txn.getTxnID().toString() +"] with unexpected state : "
```

Comment:
It's better to create the template error message for InvalidTxnStatusException, looks like using a String formater, so that you can reuse the formater.